### DISCUSSION TITLE
What to do in case of DDoS attack?

### DISCUSSION THREAD

**Discussion by angel-luis** [vercel/next.js#41485] [Help] [answered]
We've received a 160GB bandwith attack in 2 hours so, having in count that the hobby plan have 100GB, how a small website can handle several attacks during a month? Vercel doesn't have any kind of protection?

**Comment by icyJoseph** [comment]
Weird because Vercel does have DDOS protection, according to this [site](https://vercel.com/security#collapse-button-:R1hisfakmH1:):

> Yes. Vercel Enterprise customers are covered by two forms of DDoS protection. Our systems can automatically detect and block malicious attacks on customer sites. For significantly larger, distributed attacks, we work closely with the customer to ensure your site(s) stay online. The combination of automated prevention and direct communication from our Customer Success Managers helps ensure your site is resilient to attacks.

Hope your site stays up!

**Reply by heytulsiprasad** [reply]
Hey good to hear that. But small question, is this feature of DDoS prevention actually functional for pro/hobby projects or only for enterprise customers? Thanks.

**Reply by alessiomaffeis** [reply]
Same! Subscribing to this thread.

**Reply by imkarthikk** [reply]
Their DDoS prevention is on 'their' network level. Not for our accounts. They still don't admit this and this thread is an honest reflection of their way to trick users.

**Comment by xrpns** [comment]
Any update on this?

**Comment by imkarthikk** [comment]
Here's my experience with Vercel's team about this DDoS: https://twitter.com/imkarthikk/status/1616509282966704134 & https://twitter.com/imkarthikk/status/1612671311574339585

**Reply by xrpns** [reply]
I have set it up with cloudflare for now using their proxy. Under attack mode also is working, I will closely monitor this now, will report here if I find a solution, haven't suffered a DDOS attack yet but watching.

**Comment by leerob** [comment, marked as answer]
Hey, wanted to follow up here.

We just launched **[Attack Challenge Mode](https://vercel.com/changelog/prevent-malicious-traffic-with-attack-challenge-mode-for-vercel-firewall)** –an extra layer of defense for DDoS and malicious traffic, free for all plans. When enabled, the Vercel Firewall will serve a challenge page to help mitigate attacks.

Hope this helps!

**Reply by fforootd** [reply]
Nice, great to have this feature available 👍

**Reply by samxaradzenika** [reply]
how does it do against attacks using proxy ?

**Reply by leerob** [reply]
Good news, we have now launched even more powerful features with our Firewall! You can now set up custom rules to log, block, challenge, and soon rate limit traffic.

https://vercel.com/changelog/block-rate-limit-and-challenge-traffic-with-the-vercel-firewall

**Reply by imkarthikk** [reply]
@leerob I appreciate you guys shipping this feature super fast in just 2 years of this issue originating. And I also  appreciate how silent you guys get when asked about such issues on Twitter as well. Now that you have shipped this feature, can you commit on a timeline for “soon rate limit traffic”? What’s soon here? People deserve to know before they could commit their production apps on Vercel.

**Reply by leerob** [reply]
Happy to flag your team in to try things out. It should be out of private beta in the coming month.

**Reply by Osama-V2** [reply]
This really just save my website, like 1.7M request in one day, facebook spam went crazy.
